cranelift/frontend/src/switch.rs

```diff
@@ -525,13 +530,31 @@ block3:
 
 block0:
     v0 = iconst.i8 0
-    v1 = uextend.i32 v0
+    v1 = uextend.i64 v0
     v2 = icmp_imm eq v1, -1
     brnz v2, block1
     jump block4
 
 block4:
-    br_table.i32 v1, block0, jt0"
+    br_table.i64 v1, block0, jt0"
+        );
+    }
+
+    #[test]
+    fn switch_extends_small_inputs() {
+        // This is a regression test for a bug that we found where we would emit a cmp
+        // with a type that was not able to fully represent a large index.
+        //
+        // See: https://github.com/bytecodealliance/wasmtime/pull/4502#issuecomment-1191961677
+        let func = setup!(1, [0x4100_0000_00bf_d470,]);
```

Comment:
This is not a valid input IMHO. I don't think we should support cases larger than what the maximum value of the input (in this case 255 as it uses an i8). They can't be hit anyway.

Reply:
Makes sense. I'll reject these inputs instead.

Reply:
Wait, what about if we have something like this `setup!(1, [10, 0x4100_0000_00bf_d470,]);`

We are able to hit one of the outputs, does it make sense to reject that as well?

Reply:
I think so too. I think it should require all cases to be in bounds.